internal/service/verify/resource_verify_policy_test.go

```diff
@@ -0,0 +1,523 @@
+package verify_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/patrickcping/pingone-go-sdk-v2/management"
+	"github.com/patrickcping/pingone-go-sdk-v2/verify"
+	"github.com/pingidentity/terraform-provider-pingone/internal/acctest"
+	validation "github.com/pingidentity/terraform-provider-pingone/internal/verify"
+)
+
+func testAccCheckVerifyPolicyDestroy(s *terraform.State) error {
+	var ctx = context.Background()
+
+	p1Client, err := acctest.TestClient(ctx)
+
+	if err != nil {
+		return err
+	}
+
+	apiClient := p1Client.API.CredentialsAPIClient
+	ctx = context.WithValue(ctx, verify.ContextServerVariables, map[string]string{
+		"suffix": p1Client.API.Region.URLSuffix,
+	})
+
+	mgmtApiClient := p1Client.API.ManagementAPIClient
+	ctx = context.WithValue(ctx, management.ContextServerVariables, map[string]string{
+		"suffix": p1Client.API.Region.URLSuffix,
+	})
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "pingone_credential_issuer_profile" {
+			continue
+		}
+
+		_, rEnv, err := mgmtApiClient.EnvironmentsApi.ReadOneEnvironment(ctx, rs.Primary.Attributes["environment_id"]).Execute()
+
+		if err != nil {
+
+			if rEnv == nil {
+				return fmt.Errorf("Response object does not exist and no error detected")
+			}
+
+			if rEnv.StatusCode == 404 {
+				continue
+			}
+
+			return err
+		}
+
+		body, r, err := apiClient.CredentialIssuersApi.ReadCredentialIssuerProfile(ctx, rs.Primary.Attributes["environment_id"]).Execute()
```

Comment:
Likely incorrect, needs a check

Reply:
Words cannot express how annoyed I am with myself about this defect.  Resolved.